TDSv71 and other fixes

mattjdnv opened the pull request:
Re-enable TDSv71 and various translation fixes

@brianhatchl I modified some of the NodeJs translation tests and added TDSv71 ones

brianhatchl commented on translations/TDSv70.js:
```diff
@@ -33,7 +33,7 @@ hoot.require('SchemaTools');
 hoot.require('tds70_schema');
 hoot.require('tds70_rules');
-// hoot.require('config');
+hoot.require('config');
```
I see other scripts removing this commented out line.  Is it really needed here?  or should it be removed too.

mattjdnv replied:
Removed alond with some other commented out `config.xxx` lines

brianhatchl reviewed (commented):
Yes, lots of diffs to the translation tests, but is any translation behavior really changing?

mattjdnv commented:
The main behavior changes:
* Added TDSv71 tests
* Converted some of the tests into loops checking each schema
* Broke some tests out into their own files instead of being part of `TranslationServer.js`

mattjdnv commented:
I'm still trying to figure out why CircleCI is failing during `make archive`. I'm not sure if it is a timeout issue.
The weird thing is I changed the order in `Makefile.hoot` and a different schema build failed!

mattjdnv commented:
After ssh-ing into Circle CI, I think it is running out of resources when trying to run the schema generation in parallel.
I have added a 'make' target that the ci archive script can call to build the schema before trying to build the rest. I have also added a `make` job limit when the ci archive runs the target to try and  stop it running out of resources.

A better option may be to just commit the generated schema files.

mattjdnv commented:
Yes!!!
CircleCI is happy again.

brianhatchl reviewed (approved):
Nice :sunglasses:

mattjdnv commented:
Need to do some final QA then can merge.

mattjdnv commented:
Fixes:
* Fixed the default values in the TDSv71 Thematic Schema
* Fixed how the FCSUBTYPE attribute is populated when writing to a thematic layer
* Broke out a couple of tests from the TranslationServer into their own files.
